zebra-chain/src/transaction.rs

```diff
@@ -146,4 +154,33 @@ impl Transaction {
                 Some(TransparentInput::Coinbase { .. })
             )
     }
+
+    // TODO(jlusby): refine type
+    pub fn sighash(&self, network: Network, height: BlockHeight, hash_type: u32) -> Hash {
+        sighash::SigHasher {
+            trans: self,
+            network,
+            height,
+            hash_type,
+        }
+        .sighash()
```

Comment:
As noted below (https://github.com/ZcashFoundation/zebra/pull/870/files#r471234958) the `SigHasher` type has no function other than to serve as the implementation of the `Transaction::sighash()` function, by bundling its parameters into a `SigHasher` and then calling `SigHasher::sighash()` instead.

Rather than doing this, it would be better to implement the sighash functionality directly inside `Transaction::sighash`. If there's a need to factor out different chunks of code, that can be done by calling functions defined in the `sighash` module.

Reply:
https://github.com/ZcashFoundation/zebra/pull/870#discussion_r471804327

Reply:
The Network and Height are only used to derive the NetworkUpgrade.

So we can remove one parameter by taking the NetworkUpgrade, rather than the Network and Height. (This change also makes the code a bit simpler, and makes it harder to misuse Network or Height.)